pkg/database/postgres.go

```diff
@@ -0,0 +1,116 @@
+package database
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+type Postgres struct {
+	connectionString string
+	logFile          string
+}
+
+func NewPostgres(connectionStr, logFile string) (Postgres, error) {
+	readyCmd := pgIsReadyCmd(connectionStr, 30)
+	errorBuff := captureError(readyCmd)
+
+	err := readyCmd.Run()
+	if err != nil {
+		return Postgres{}, errors.Wrapf(err, "failed to access database, error output: %s", errorBuff.String())
+	}
+
+	return Postgres{connectionString: connectionStr, logFile: logFile}, nil
+}
+
+func (p Postgres) Export(path string) error {
```

Comment:
I'm just curious why this receiver (and the others in this file) isn't a pointer? I'm not sure I've actually ever seen a method defined with a concrete receiver and I'm not sure I understand how it will affect the behavior

Reply:
The difference is that modifying anything the `Postgres` struct would not persist after the method exist as it does not operate on "original" object.
I like to use value receivers where they are sufficient (no need to modify any struct fields) as this seems more explicit to me, that this method does not change the receiver.